### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by hemantso**
- Create a book and remove an action created
- actions connected with reducer
-  redux state updated using actions
- displaying data of reducer state on components

### CODE REVIEW

**Comment by sumancrest0001** [comment]
### Status: Changes Required :gear:

**Nice work @hemant-soni-vst-au4** :clap: 
You have implemented all the requirements for this milestone. Users can easily add and remove books from the store. But there is a minor change required before moving to the next milestone. Let's have a look at it.

File: src/containers/BooksForm.js [hunk 1]
```diff
@@ -1,27 +1,69 @@
...
+    });
+  };
+
+  const handleSelectChange = event => {
+    setstate({
+      ...state,
+      category: event.target.value,
+    });
+  };
+
```
Comment (sumancrest0001) [inline src/containers/BooksForm.js]: [Required] We can remote one of these functions as they perform the same task. You can pass the name and value argument to a function and update the state according to the name.
For example,
const handleChange = (name, value) => {
name: value 
}
something like this.

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -1,27 +1,69 @@
...
+  const handleSubmit = () => {
+    if (state.title === '') return;
```
Comment (sumancrest0001) [inline src/containers/BooksForm.js:30]: [Required] We can prevent the default form submission event using e.preventDefault(). [Reference](https://www.w3schools.com/jsref/event_preventdefault.asp)

**Review by sumancrest0001 — CHANGES_REQUESTED** [review]
**Please make the changes and submit for another code review.**

**If you have any queries about the code review, please message me on Slack.
Suman Shrestha**

**Happy coding !!!**

**Comment by sumancrest0001** [comment]
[Required] I forgot to mention. Please capitalize your commit messages. [Reference](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)

**Comment by hemantso** [comment]
- updated event handler and follow the DRY instruction
- made the required changes

File: src/containers/BooksForm.js [hunk 3]
```diff
@@ -1,27 +1,64 @@
...
+      ...state,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = () => {
+    if (state.title === '') return;
+    createBook({
+      ...state,
+      id: Math.floor(Math.random() * 100),
```
Comment (mosaaleb) [inline src/containers/BooksForm.js]: - [x] Depending in `random` to generate unique ids doesn't guarantee uniqueness. I got this console error when I tried to add a new book which happens to generate the same id for another book.

Screenshot from 2021-01-16 17-03-01

I suggest that you depend on another technique like `Date.now()` which will always guarantee a unique ID or use [uuid](https://www.npmjs.com/package/uuidv4) library.

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Changes required

Thanks @hemant-soni-vst-au4 for implementing the changes required from the previous review. However, there is a console error I have encountered while trying to create a new book. Kindly check my comments below:

- [x] As requested from the previous review, let's use the recommended git commit message [style](https://commit.style/). 

Kindly handle these requirements and submit a new code review.

##### Muhammad Ebeid

**Review by Forison — APPROVED** [review]
# Status::Approved✔️
## Good job on addressing the issues raised in the previous review, console is clear and your key guarantees uniqueness. Go ahead and merge the branch and proceed to the next milestone👏👏👏
#### Happy coding
Addo Forison

**Pull request merged**